twingate/resource_group.go

```diff
@@ -40,40 +40,36 @@ func resourceGroup() *schema.Resource {
 func resourceGroupCreate(ctx context.Context, resourceData *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	client := meta.(*Client)
 
-	groupName := resourceData.Get("name").(string)
-	group, err := client.createGroup(ctx, graphql.String(groupName))
-
+	group, err := client.createGroup(ctx, graphql.String(resourceData.Get("name").(string)))
 	if err != nil {
 		return diag.FromErr(err)
 	}
 
-	resourceData.SetId(group.ID.(string))
-	log.Printf("[INFO] Group %s created with id %s", groupName, resourceData.Id())
-
-	waitForResourceAvailability()
+	log.Printf("[INFO] Group %s created with id %v", group.Name, group.ID)
 
-	return resourceGroupRead(ctx, resourceData, meta)
+	return resourceGroupReadHelper(resourceData, group, nil)
 }
 
 func resourceGroupUpdate(ctx context.Context, resourceData *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	client := meta.(*Client)
 
-	groupName := resourceData.Get("name").(string)
+	var err error
 
-	if resourceData.HasChange("name") {
-		groupID := resourceData.Id()
+	group := &Group{
+		ID:   graphql.ID(resourceData.Id()),
+		Name: graphql.String(resourceData.Get("name").(string)),
+	}
 
-		err := client.updateGroup(ctx, graphql.ID(groupID), graphql.String(groupName))
+	if resourceData.HasChange("name") {
```

Comment:
Why do we need this check?
If TF calls `resourceGroupUpdate` it means something in the group changed right? (and since we don't have any other property than `name` ...)

Reply:
right, we can simplify it